src/main/java/uk/ac/ebi/atlas/experimentpage/markergenes/MarkerGenesDao.java

```diff
@@ -17,11 +17,19 @@ public MarkerGenesDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
     }
 
     private static final String SELECT_MARKER_GENES_WITH_AVERAGES_PER_CLUSTER =
```

Comment:
I am afraid that I cannot get the data by this DAO query. Please have a double check and do refactor.

Reply:
```
SELECT
g.experiment_accession,
m.gene_id,
g.variable as k_where_marker,
h.value as cluster_id_where_marker,
g.value as cluster_id,
g.id as group_id,
m.marker_probability as marker_p_value,
s.mean_expression, s.median_expression
FROM
scxa_cell_group_marker_gene_stats s,
scxa_cell_group_marker_genes m,
scxa_cell_group g,
scxa_cell_group h
WHERE
s.cell_group_id=g.id and
s.marker_id=m.id and
m.cell_group_id = h.id and g.experiment_accession='E-MTAB-6505' and
m.marker_probability < 0.05 and
g.variable = '10' and
expression_type=0 order by m.marker_probability;
```

Reply:
Try this query on the `test ` environment, you will see the result. Choose experiment accession from the `scxa_cell_group` table and k value(variable), then pass those inputs to the above query to see results.

Reply:
Thanks for the reference query. I got confused with the `value` and `variable` at first.